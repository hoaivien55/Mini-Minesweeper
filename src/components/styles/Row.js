import styled from "styled-components";

const Row = styled.div`
	display: flex;
	justify-content: ${(props) => props.justifyContent || "center"};
    align-items: ${(props) => props.alignItems || "center"};
    padding: ${(props) => props.padding || "0px"};
`;

export default Row;