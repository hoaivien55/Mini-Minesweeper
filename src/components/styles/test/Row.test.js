import renderer from "react-test-renderer";
import Row from "../Row";

it("Row: Virtual DOM snapshot", () => {
	const tree = renderer.create(<Row />).toJSON();
	expect(tree).toMatchSnapshot();
});

it("Row: change props", () => {
	const tree = renderer
		.create(<Row justifyContent="start" alignItems="end" padding="10px" />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
