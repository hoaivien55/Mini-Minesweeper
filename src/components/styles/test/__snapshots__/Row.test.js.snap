// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row: Virtual DOM snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px;
}

<div
  className="c0"
/>
`;

exports[`Row: change props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
  padding: 10px;
}

<div
  className="c0"
/>
`;
